### DISCUSSION TITLE
TinyMCE plugins

### DISCUSSION THREAD

**Discussion by bcartier** [directus/directus#12327] [Archive]
Is it possible to initialize open source TinyMCE plugins using the **Option Override** field config? Or would this require building a custom Interface? 

Specifically, I was hoping to enable [charmap](https://www.tiny.cloud/docs/plugins/opensource/charmap/)

**Comment by rijkvanzanten** [comment]
The list of loaded plugins is currently hardcoded:

https://github.com/directus/directus/blob/3043fafb64eefc714a4aa147ef3a37245ee814f5/app/src/interfaces/input-rich-text-html/input-rich-text-html.vue#L170-L184

so just adding the plugin loading in the option overrides wouldn't be enough. The ideal situation here would be to update the interface to allow any core tinymce plugin to be loaded and configured, so it becomes a dedicated interface option. As of right now, a custom interface would be the only way to achieve that.

**Reply by bcartier** [reply]
Thanks Rijk!

**Reply by CapitaineToinon** [reply]
Is there any reason not to include all free plugins since we can already configure from the dashboard which plugin should be displayed or not already?

**Reply by rijkvanzanten** [reply]
> Is there any reason not to include all free plugins since we can already configure from the dashboard which plugin should be displayed or not already?

Size / performance. TinyMCE registers all plugins globally, so loading them all by default would download a huge amount of (unused) javascript

**Reply by Michael-Nussbaumer** [reply]
If I want to have a custom plugin for example to pick links from our url shortener, I would need to make copy of the existing interface, right?

**Reply by rijkvanzanten** [reply]
That's correct

**Reply by jayakornk** [reply]
I'm wondering about how to actually copy `input-rich-text-html` interface and make a custom interface extension as this interface import some hook that doesn't exist in `@directus/extensions-sdk`. Any suggestion? (I would love to add Anchor plugin).

**Reply by quentinglorieux** [reply]
Anyone has worked on this ? Like an directus extension with a customizable TinyMCE instance ?
I just blindly copied the interface from the directus repo but then how to correctly connect all the import for building ?
(linking all the dependencies)

**Reply by quentinglorieux** [reply]
Anyone maybe ?
